src/graph.cpp

```diff
@@ -160,26 +155,42 @@ void Graph::kruskal(){
  * @param candidate - Vertex that is going to be checked
  * @return true if the candidate vertex is adjacent to the tour's last vertex, false otherwise
  */
-bool Graph::isConnectable(Vertex *candidate) const {
-    for (Edge *e : (*tour.second.rend())->getAdj()){
+bool Graph::isConnectable(std::shared_ptr<Vertex> &candidate) const {
+    for (std::shared_ptr<Edge> &e : (*tour.course.rbegin())->getAdj()){
         if (e->getDest() == candidate) return true;
     }
     return false;
 }
 
+ /**
+  * Adds a vertex to the tour structure and updates the total distance
+  * Time Complexity: O(|E|) (average case) | O(|V|*|E|) (worst case)
+  * @param stop - Vertex to add
+  */
+void Graph::addToTour(std::shared_ptr<Vertex> &stop) {
+    if (!tour.course.empty()){
+        tour.distance += findEdge((*(tour.course).rbegin())->getId(), stop->getId())->getLength();
+    }
+    tour.course.push_back(stop);
+}
+
+/**
+ * Adds the distance between the first and last nodes so the tour "closes"
+ * Time Complexity: O(|E|) (average case) | O(|V|*|E|) (worst case)
+ */
+void Graph::tieDownTour(){
+    tour.distance += findEdge((*tour.course.begin())->getId(), (*tour.course.rbegin())->getId())->getLength();
```

Comment:
I think not only should you add here the distance, but also the edge to the tour course, it is part of it afterall

Reply:
Fixed